python/lsst/ts/attcpip/csc.py

```diff
@@ -225,35 +234,53 @@ async def start_clients(self) -> None:
             cmd_evt_port = self.simulator.cmd_evt_server.port
             telemetry_port = self.simulator.telemetry_server.port
 
-        self.cmd_evt_client = tcpip.Client(
-            host=host, port=cmd_evt_port, log=self.log, name="CmdEvtClient"
-        )
-        self.telemetry_client = tcpip.Client(
-            host=host, port=telemetry_port, log=self.log, name="TelemetryClient"
-        )
+        # Stop the clients in case the configuration has changed.
+        if self.cmd_evt_client.connected and (
+            self.cmd_evt_client.host != host or self.cmd_evt_client.port != cmd_evt_port
+        ):
+            await self._stop_cmd_evt_task_and_client()
+        if self.telemetry_client.connected and (
+            self.telemetry_client.host != host
+            or self.telemetry_client.port != telemetry_port
+        ):
+            await self._stop_telemetry_task_and_client()
```

Comment:
Why not stop both if anyone of clients failed? Actually, I thought if one client failed, the other should be as well.

Reply:
The comment above the first client specifies "Stop the clients in case the configuration has changed.". So this is done only when the configuration has changed, not when the connection has failed. That is taken care of in lines 248 to 262.